### DISCUSSION TITLE
Queryset in ObjectVar in script

### DISCUSSION THREAD

**Discussion by xeor** [netbox-community/netbox#12480] [Help Wanted!] [answered]
I want to make an `ObjectVar` that bases its content out of a queryset.
In short, I want it to have a list of `CustomField`, but only those fields available in `ipam.prefix`

Is this even possible using `query_params`?

The fields I want is from `CustomField.objects.filter(content_types__in=ContentType.objects.filter(app_label='ipam', model='prefix'))`

I have tried to override queryset in ObjectVar, but it doesnt work, probably because it's a task-runner that does the actual execution.

**Comment by xeor** [comment]
I found a solution to do this;

```py
    def custom_field_choices():
        queryset = CustomField.objects.filter(content_types__in=ContentType.objects.filter(app_label='ipam', model='prefix')).order_by('name')
        return [(cf.name, str(cf)) for cf in queryset]

    field = ChoiceVar(
        choices=custom_field_choices(),
        label='Field',
        required=True
    )
```

**Reply by kkthxbye-code** [reply, marked as answer]
This should work for an ObjectVar.

`query_params={ "content_types": "ipam.prefix" }`

**Reply by xeor** [reply]
Thanks! I tested this, and this works as well!
